timestamp-impl/src/main/java/com/palantir/timestamp/PersistentUpperLimit.java

```diff
@@ -23,28 +23,26 @@
 import com.palantir.common.time.SystemClock;
 
 public class PersistentUpperLimit {
+
     private final TimestampBoundStore tbs;
     private final Clock clock;
+    private final TimestampAllocationFailures allocationFailures;
+
     @GuardedBy("this")
     private volatile long cachedValue;
     private volatile long lastIncreasedTime;
 
-    public PersistentUpperLimit(TimestampBoundStore tbs, Clock clock) {
+    public PersistentUpperLimit(TimestampBoundStore tbs, Clock clock, TimestampAllocationFailures allocationFailures) {
         this.tbs = tbs;
         this.clock = clock;
+        this.allocationFailures = allocationFailures;
```

Comment:
In this specific case mainly because we don't want to NPE in the failure handling, but in general we need to be better at verifying preconditions on construction